File: app/src/main/java/com/monk/retrofit/mvp/model/RetrofitModleImpl.java
package com.monk.retrofit.mvp.model;

import com.monk.retrofit.netapi.HttpMethods;

import io.reactivex.Observable;
import okhttp3.ResponseBody;

/**
 * @author monk
 * @date 2019-02-19
 */
public class RetrofitModleImpl implements RetrofitModle {

    @Override
    public Observable<ResponseBody> getData(String unitGuid) {
        return HttpMethods.getInstance().getHttpApi().getDataForMap("", "", unitGuid);
    }
}

File: module_fragment/src/main/java/com/monk/modulefragment/retrofit/mvp/model/RetrofitModleImpl.kt
package com.monk.modulefragment.retrofit.mvp.model

import com.monk.modulefragment.retrofit.netapi.HttpMethods
import io.reactivex.Observable
import okhttp3.ResponseBody

/**
 * @author monk
 * @date 2019-02-19
 */
class RetrofitModleImpl : RetrofitModle {
    override fun getData(unitGuid: String?): Observable<ResponseBody?> {
        return HttpMethods.getInstance().httpApi.getDataForMap("", "", unitGuid)!!
    }
}